labelbox/orm/db_object.py

```diff
@@ -244,3 +254,28 @@ def delete(self):
         a call to this you should not use this DB object anymore.
         """
         type(self).bulk_delete([self])
+
+
+def beta(fn):
+
+    def wrapper(self, *args, **kwargs):
+        if not isinstance(self, DbObject):
+            raise TypeError(
+                "Cannot decorate functions that are not functions of `DbOjects` with `beta` decorator"
+            )
+        if not self.client.enable_beta:
+            raise Exception(
+                f"This function {fn.__name__} relies on a beta feature in the api. This means that the interface could change."
+                " Set `enable_beta=True` in the client to enable use of these functions."
+            )
+        execute_fn = self.client.execute
+        try:
+            self.client.execute = partial(execute_fn, beta=True)
```

Comment:
the decision to use the beta endpoint is determined by the parameter in the execute function. This is so that we can support calling beta endpoints from paginated collections.

Reply:
This has been removed because it wasn't thread safe